src/libstore/ipfs-binary-cache-store.cc

```diff
@@ -209,32 +214,26 @@ void IPFSBinaryCacheStore::sync()
     if (trustless)
         return;
 
-    if (!optIpnsPath) {
-        if (initialIpfsPath != state->ipfsPath)
-            throw Error(
-                "You performed store-modifying actions, creating a new store whose IPFS address doesn't match the configured one:\n"
-                "  configured: %s\n"
-                "  modified: %s\n"
-                "\n"
-                "This happens when one has configured nix to use a store via an IPFS hash. Since the store is immutable a new one is made (functional update) and the \"modified\" is its hash. Nix isn't going to statefully switch to using that hash, however, because that would be in violation of the configuration Nix has been given.\n"
-                "\n"
-                "You can change you configuration to use this hash, and run the command again in which case it will succeed with a no-opt. But if you are going to modify the store on a regular basis you should use DNS-link or IPNS instead so you have a properly mutable store, and avoid getting this message again.",
-                formatPathAsProtocol(initialIpfsPath), formatPathAsProtocol(state->ipfsPath));
-        else
-            return;
-    }
+    if (state->ipfsPath == initialIpfsPath)
+        return;
```

Comment:
```suggestion
        return;
    // If we aren't in trustless mode (handled above) and we don't allow modifications, state->ipfsPath should never be changed from the initial one
    assert(allowModify);
```
TODO wrap the comment for line lenth etc.